pennylane/operation.py

```diff
@@ -873,11 +875,55 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
                 f"{len(self._wires)} wires given, {self.num_wires} expected."
             )
 
+        self._check_batching(params)
+
         self.data = list(params)  #: list[Any]: parameters of the operator
 
         if do_queue:
             self.queue()
 
+    def _check_batching(self, params):
+        """Check if the expected numbers of dimensions of parameters coincides with the
+        ones received and sets the ``_batch_size`` attribute.
+
+        Args:
+            params (tuple): Parameters with which the operator is instantiated
+
+        The check always passes and sets the ``_batch_size`` to ``None`` for the default
+        `Operator.ndim_params` property but subclasses may overwrite it to define fixed
+        expected numbers of dimensions, allowing to infer a batch size.
+        """
+        self._batch_size = None
+        try:
+            ndims = tuple(qml.math.ndim(p) for p in params)
```

Comment:
Would this give odd results for e.g., `QubitUnitary` or `QubitStateVector`?

Reply:
It should output `(2,)` for unbatched input to `QubitUnitary` and `(3,)` for batched input.
For `QubitStateVector` it would be `(1,)` or `(2,)`, respectively.